console/qscan/qscan.cpp

```diff
@@ -9,841 +9,932 @@
-
-		printf("\nDevice Read/Write capabilities:\n");
-		for (int idx=0; rw_capabilities[idx].id || strlen(rw_capabilities[idx].name); idx++) {
-			if (rw_capabilities[idx].id)
-				printf(CAPMEDIA "%11s: %c%c\n", rw_capabilities[idx].name,
-					(dev->rd_capabilities & rw_capabilities[idx].id) ? 'R' : '-',
-					(dev->wr_capabilities & rw_capabilities[idx].id) ? 'W' : '-');
-		}
-		if (!(flags & FL_MINFO))
-			goto end;
+    printf(CAPVND "Hide CD-R         : %s\n",
+           dev->ven_features & PX_HCDRSS ? (dev->plextor.hcdr ? "ON" : "OFF")
+                                         : "-");
+    printf(CAPVND "SingleSession     : %s\n",
+           dev->ven_features & PX_HCDRSS ? (dev->plextor.sss ? "ON" : "OFF")
+                                         : "-");
+    printf(CAPVND "SpeedRead         : %s\n",
+           dev->ven_features & PX_SPDREAD
+               ? (dev->plextor.spdread ? "ON" : "OFF")
+               : "-");
+    printf(CAPVND "Silent mode       : %s\n",
+           dev->ven_features & PX_SILENT
+               ? (dev->plextor_silent.state ? "ON" : "OFF")
+               : "-");
+    if ((dev->ven_features & PX_SILENT) && (dev->plextor_silent.state))
+      plextor_print_silentmode_state(dev);
+    printf(CAPVND "SecuRec           : %s\n",
+           dev->ven_features & PX_SECUREC
+               ? (dev->plextor.securec ? "ON" : "OFF")
+               : "-");
+    printf(CAPVND "PoweRec           : %s\n",
+           dev->ven_features & PX_POWEREC
+               ? (dev->plextor.powerec_state ? "ON" : "OFF")
+               : "-");
+    if (dev->ven_features & PX_POWEREC) {
+      if (dev->media.type & DISC_CD)
+        printf(CAPVND "    PoweREC Speed : %dX (CD)\n",
+               dev->plextor.powerec_spd / 176);
+      if (dev->media.type & DISC_DVD)
+        printf(CAPVND "    PoweREC Speed : %dX (DVD)\n",
+               dev->plextor.powerec_spd / 1385);
+    }
+    printf(CAPVND "GigaRec           : %s\n",
+           dev->ven_features & PX_GIGAREC ? "YES" : "-");
+    if (dev->ven_features & PX_GIGAREC)
+      print_gigarec_value(dev);
+    printf(CAPVND "VariRec CD        : %s\n",
+           dev->ven_features & PX_VARIREC_CD
+               ? (dev->plextor.varirec_state_cd ? "ON" : "OFF")
+               : "-");
+    if ((dev->ven_features & PX_VARIREC_CD) && (dev->plextor.varirec_state_cd))
+      print_varirec(dev, VARIREC_CD);
+    if (dev->wr_capabilities & DEVICE_DVD) {
+      printf(CAPVND "VariRec DVD       : %s\n",
+             dev->ven_features & PX_VARIREC_DVD
+                 ? (dev->plextor.varirec_state_dvd ? "ON" : "OFF")
+                 : "-");
+      if ((dev->ven_features & PX_VARIREC_DVD) &&
+          (dev->plextor.varirec_state_dvd))
+        print_varirec(dev, VARIREC_DVD);
+      printf(CAPVND "DVD+R bitset      : %s\n",
+             dev->ven_features & PX_BITSET_R
+                 ? (dev->book_plus_r ? "DVD-ROM book" : "DVD+R book")
```

Comment:
Codacy found an issue: [Clarify calculation precedence for '&' and '?'.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)